Build/SaveLoadScreen.cc

```diff
@@ -327,7 +349,8 @@ static void LeaveSaveLoadScreen()
 	ScreenID const exit_screen =
 		gfCameDirectlyFromGame                     ? guiPreviousOptionScreen :
 		guiPreviousOptionScreen == MAINMENU_SCREEN ? MAINMENU_SCREEN         :
-		OPTIONS_SCREEN;
+		guiPreviousOptionScreen == GAME_INIT_OPTIONS_SCREEN ? GAME_INIT_OPTIONS_SCREEN : // Added to handle interaction during Dead is Dead game start
+		guiPreviousOptionScreen == INTRO_SCREEN ? INTRO_SCREEN : OPTIONS_SCREEN;
```

Comment:
What a monster statement! This would benefit from a refactor to a set + a conditional switch on guiPreviousOptionScreen.